daemon/logger/awslogs/cloudwatchlogs.go

```diff
@@ -192,8 +219,50 @@ func (l *logStream) Close() error {
 	return nil
 }
 
-// create creates a log stream for the instance of the awslogs logging driver
+// create creates log group and log stream for the instance of the awslogs logging driver
 func (l *logStream) create() error {
+	if err := l.createLogStream(); err != nil {
+		if l.logCreateGroup {
+			if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == resourceNotFoundCode {
```

Comment:
Is `resourceNotFoundCode` specific enough? Are there other possible reasons this code is returned? Alternatively,  should we check if the log-group exists, and if not create it (I realise this would give a small performance penalty as well)